### ISSUE TITLE
Background Compression of Backups

### ISSUE THREAD

**Issue by Kaiede** [Kaiede/Bedrockifier#23]
One issue with performance around live backups is that as worlds get bigger (especially on Java), having the Minecraft server unable to write to disk for long periods of time seems to present lag to users connected to the server. But it may also be the compression algorithm itself that’s the problem. So not 100% certain what the actual source of lag is here yet. 

One possible way to improve the experience here would be to make a local copy of the world first to a separate folder.  Once the copy is complete, the server can be unlocked which may allow the players to not have to deal with the lag spikes. This would require more disk space to perform the backups though.

Need to test this on small scale first to see how much benefit there is and if it allows backups to be done live on Java.

**Comment by Kaiede** [comment]
Branch has spun up with first set of changes, with testing done on an M1 Mac mini in an Ubuntu VM with a world that's about 2.28GB in size.

Without Defer Compression:
- 48s total
- User impression was nonexistent (breaking blocks and picking up items)

With Defer Compression:
- 30s for the file copy
- 12s for the compression
- User impression is that there was still a couple hiccups with lag when attempting to pick up items on the floor or attacking a skeleton. 

So the test seems to suggest that the lag spikes themselves are somewhat conditional on the state of the server itself. Interesting. Deferring the compression is still rather beneficial for minimizing how long the server has to go with auto-saving disabled, but it still doesn't guarantee that users won't see spikes (Bedrock still seems more performant here).

**Comment by Kaiede** [comment]
Another possibility here is to do the actual backup at low priority so that the server gets priority. This could further lengthen the time required to do the backup, and so needs a round of testing with a different branch.

**Comment by Kaiede** [comment]
Tried running a version that did backups at background priority. 

- 31s total for the backup (why faster?)
- Server still reported that it was running 9.4s behind, 18s after the backup had fully completed. 

So, I guess the question is why is the server overloaded here?
